tests/testthat/test-smoke-tests.R

```diff
@@ -0,0 +1,21 @@
+library(testthat)
+library(hydromad)
+
+context("Smoke tests")
+
+data(Cotter)
+ts90s <- window(Cotter$Q, start = "1990-01-01", end = "1999-12-31")
+
+## Test that tsdiag.hydromad returns correct class
+test_that("can plot arima models with tsdiag", {
```

Comment:
These examples don't actually trigger tsdiag.hydromad - none of the first arguments are hydromad objects

Reply:
I think we need to check how these functions are actually used in the code
It seems there is no exported way of creating an ltv object and tsdiag.ltv is not exported either
```
U <- ts(rpois(200, lambda=0.25))
Q <- filter(0.5 * U, filter=0.5, method="recursive")
Q <- Q + rnorm(200, sd=0.1)
Q <- pmax(Q, 0)
foo <- hydromad:::ltv(Q, U, na=1, nb=1, nc=1)
hydromad:::tsdiag.ltv(foo)
```

The hydromad method relies on a $uh being added to the object and I haven't yet tracked down which function does that.

Reply:
Sorry to do this to you, but my thought is now to remove the testing code from this PR and add an issue. I'm satisfied that the changes above are correct.

Reply:
I went to xyplot.hydromad.Rd to grab an example for testing:

```
data(Canning)
cannCal <- window(Canning, start = "1978-01-01", end = "1982-12-31")
mod <-
  hydromad(cannCal, sma = "cwi", tw = 162, f = 2, l = 300,
           t_ref = 0, scale = 0.000284,
           routing = "expuh", tau_s = 4.3, delay = 1, warmup = 200)
```

Unless I'm mistaken, `mod` should be a hydromad object? I tried a couple of things:

1. `hydromad::tsdiag(arima(mod))` returns `Error in arima(mod) : 'x' must be numeric` since `arima` only accepts a univariate time series. 

2. `hydromad::tsdiag(arima(mod$data$P))` returns a plot.

3. `hydromad::tsdiag(mod)` returns `Error in sqrt(object$sigma2) : non-numeric argument to mathematical function`.

I feel like I'm missing something here. Am I using an incorrect type of hydromad object to test this out, or is it incorrect to try and force a hydromad object to an arima model?

Reply:
Ah I posted my comment before refreshing the pages, so I didn't see that you'd made a new comment.

Reply:
No worries, I'll taken out the tests.

Reply:
Sorry - I've been commenting between other tasks and therefore a bit adhoc
`tsdiag(mod)` should work, but as you found out, it only works for some hydromad objects and I'm not sure which yet.